### PR TITLE
Task6 redux up

### PR DESCRIPTION
**Pull request by FallenBY1**

### CODE REVIEW

File: movieapp/src/containers/FilterElement/FilterElement.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    value: 'comedy'
+  },
+  {
+    id: '3',
+    title: 'Animation',
+    value: 'animation'
+  }
+];
+
+export function FilterElement(): JSX.Element {
```
Comment (DrDiman) [inline movieapp/src/containers/FilterElement/FilterElement.tsx]: do not use postfixes naming as `*Element`, `*Array` it's antipattern

File: movieapp/src/containers/FilterElement/FilterElement.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const dispatch = useDispatch();
+
+  function FilterMovies(e: any): void {
+    fetch('http://localhost:4000/movies?searchBy=genres&filter=' + e.currentTarget.value + '&sortOrder=asc')
```
Comment (DrDiman) [inline movieapp/src/containers/FilterElement/FilterElement.tsx]: creation of the query string should be as pure function from utils folder.

e.g. 
```
function searchParamsToQueryString (searchParams): String => {}

```
usage: 
```
const query = searchParamsToQueryString({
  searchBy: 'genres',
  sortOrder: 'asc'
}) // '?searchBy=genres&sortOrder=asc'
```

File: movieapp/src/containers/SortElement/SortElement.tsx [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
     id: '2',
     title: 'Release date',
-    link: 'release-date'
+    value: 'release_date'
   },
   {
     id: '3',
     title: 'Rating',
-    link: 'rating'
+    value: 'vote_average'
```
Comment (DrDiman) [inline movieapp/src/containers/SortElement/SortElement.tsx]: all of this values are constants (belong to enum e.g. 
```
const SortCriteria = {
  RELEASE_DATE: 'release_date', 
  VOTE_COUNT: 'vote_count', 
  ...
}
```

File: movieapp/src/containers/SortElement/SortElement.tsx [hunk 4]
```diff
@@ -1,29 +1,43 @@
...
   }
 ];
 
 export function SortElement(): JSX.Element {
+  const dispatch = useDispatch();
+
+  function SortMovies(e: any): void {
+    fetch('http://localhost:4000/movies?sortBy=' + e.currentTarget.value + '&sortOrder=asc')
```
Comment (DrDiman) [inline movieapp/src/containers/SortElement/SortElement.tsx]: u have installed axios, but u don't use it 0_o

File: movieapp/src/hooks/MovieContext.tsx [hunk 5]
```diff
@@ -1,44 +1,43 @@
...
-  const [newMovies, setNewMovies] = useState<IMovie[]>(newMoviesMock);
-  const value = useMemo(() => ({ newMovies, setNewMovies }), [newMovies]);
+  const newMovies = useSelector((state: any) => state.movies);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    fetch('http://localhost:4000/movies')
+      .then((res) => res.json())
+      .then((movies) => dispatch(receiveMoviesFromApiSuccess(movies.data) as AnyAction))
+      .catch((error) => dispatch(receiveMoviesFromApiError(error) as AnyAction));
```
Comment (DrDiman) [inline movieapp/src/hooks/MovieContext.tsx:20]: u have installed thunk - but u do not use it - why?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "formik": "^2.2.9"
+  }
+}
```
Comment (DrDiman) [inline package.json]: ?

File: package-lock.json [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      "resolved": "https://registry.npmjs.org/tiny-warning/-/tiny-warning-1.0.3.tgz",
+      "integrity": "sha512-lBN9zLN/oAf68o3zNXYrdCt1kP8WsiGW8Oo2ka41b2IM5JL/S1CTyX1rW0mb/zSuJun0ZUrDxx4sqvYS2FWzPA=="
+    },
+    "node_modules/tslib": {
+      "version": "1.14.1",
+      "resolved": "https://registry.npmjs.org/tslib/-/tslib-1.14.1.tgz",
+      "integrity": "sha512-Xni35NKzjgMrwevysHTCArtLDpPvye8zV/0E4EyYn43P7/7qvQwPh9BGkHewbMulVntbigmcT7rdX3BNo9wRJg=="
+    }
+  }
+}
```
Comment (DrDiman) [inline package-lock.json]: ?

File: movieapp/src/reducers/movies.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      };
+    case RECEIVE_MOVIES_FROM_API_ERROR:
+      return {
+        ...state,
+        error: action.payload.error
+      };
+    case SORT_MOVIES_BY_ID:
+      return {
+        ...state,
+        movies: [action.payload]
```
Comment (DrDiman) [inline movieapp/src/reducers/movies.ts]: something strange

File: movieapp/src/actions/actionTypes.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export const RECEIVE_MOVIES_FROM_API_SUCCESS = 'RECEIVE_MOVIES_FROM_API_SUCCESS';
+export const RECEIVE_MOVIES_FROM_API_ERROR = 'RECEIVE_MOVIES_FROM_API_ERROR';
+export const SORT_MOVIES_BY_ID = 'SORT_MOVIES_BY_ID';
```
Comment (DrDiman) [inline movieapp/src/actions/actionTypes.ts:3]: it's possible not to create these constants

with npm module called `redux-actions`

File: movieapp/src/containers/MovieList/MovieList.tsx [hunk 10]
```diff
@@ -32,13 +34,13 @@ export function MovieList(props: any): any {
...
     (id: string): void => {
-      props.setExpandedMovie(newMovies.find((movie: MovieType) => movie.id === id));
+      props.setExpandedMovie(newMovies.movies.find((movie: MovieType) => movie.id === id));
     },
     [props, newMovies]
   );
 
   function openModal(id: string): void {
-    setCurrentMovie(newMovies.filter((movie: MovieType) => movie.id === id)[0]);
+    setCurrentMovie(newMovies.movies[0].filter((movie: MovieType) => movie.id === id)[0]);
```
Comment (DrDiman) [inline movieapp/src/containers/MovieList/MovieList.tsx]: `find` method


and why is `movies[0].` used??

File: movieapp/src/containers/SortElement/SortElement.tsx [hunk 11]
```diff
@@ -1,29 +1,43 @@
...
+      .then((res) => res.json())
+      .then((movies) => {
+        dispatch(sortMoviesById(movies.data) as AnyAction);
+      })
+      .catch((error) => dispatch(receiveMoviesFromApiError(error) as AnyAction));
+  }
+
   return (
-    <select>
+    <select onChange={SortMovies}>
```
Comment (DrDiman) [inline movieapp/src/containers/SortElement/SortElement.tsx]: handlers should be called with lowercase - it's just function (not class or enum)

File: movieapp/src/hooks/MovieContext.tsx (hunk 5 above)
Comment (FallenBY1) [reply movieapp/src/hooks/MovieContext.tsx:20]: move fetch to thunk

File: movieapp/src/containers/FilterElement/FilterElement.tsx (hunk 1 above)
Comment (FallenBY1) [reply movieapp/src/containers/FilterElement/FilterElement.tsx]: Fixed

File: movieapp/src/containers/FilterElement/FilterElement.tsx (hunk 2 above)
Comment (FallenBY1) [reply movieapp/src/containers/FilterElement/FilterElement.tsx]: Fixed

File: movieapp/src/containers/SortElement/SortElement.tsx (hunk 3 above)
Comment (FallenBY1) [reply movieapp/src/containers/SortElement/SortElement.tsx]: Fixed

File: movieapp/src/containers/SortElement/SortElement.tsx (hunk 4 above)
Comment (FallenBY1) [reply movieapp/src/containers/SortElement/SortElement.tsx]: fixed

File: package.json (hunk 6 above)
Comment (FallenBY1) [reply package.json]: fixed

File: package-lock.json (hunk 7 above)
Comment (FallenBY1) [reply package-lock.json]: fixed

File: movieapp/src/containers/SortElement/SortElement.tsx (hunk 11 above)
Comment (FallenBY1) [reply movieapp/src/containers/SortElement/SortElement.tsx]: fixed

File: movieapp/src/containers/MovieList/MovieList.tsx (hunk 10 above)
Comment (FallenBY1) [reply movieapp/src/containers/MovieList/MovieList.tsx]: fixed

File: movieapp/src/reducers/movies.ts (hunk 8 above)
Comment (FallenBY1) [reply movieapp/src/reducers/movies.ts]: fixed

**Pull request merged**